Automattic/wp-calypso
# Fix some TypeScript issues across various files

## Proposed Changes

There are a ton of TypeScript errors in trunk currently. I believe this is preventing our TS checker from finding new errors, resulting in more errors being added over time. 

This diff fixes some of the errors. I tried to only make changes that were unambiguously trivial rather than risk making changes to functionality.

## Errors I could not fix here

- A bunch of errors relating to the differing definitions of the `Campaign` type introduced by https://github.com/Automattic/wp-calypso/pull/76836.
- `createHigherOrderComponent` errors in `withSitesSorting` added by https://github.com/Automattic/wp-calypso/pull/67881.
- `createHigherOrderComponent` errors in `searchSites` added by https://github.com/Automattic/wp-calypso/pull/67882.
- A pretty serious-looking `useStepNavigation` type error on the first parameter in the stepper introduced in https://github.com/Automattic/wp-calypso/pull/77780.
- An error in the dispatched type introduced by https://github.com/Automattic/wp-calypso/pull/77892.
- An error with the properties passed to `Tooltip` added by https://github.com/Automattic/wp-calypso/pull/78043.

## Testing Instructions

I think the only way is to go over each of my changes statically and make sure they make sense.

Run `yarn typecheck` on the repo to see remaining errors. `trunk` currently has 29 errors. After this PR we're down to 9 errors.
